From 003b25f39ddbd8b1324d8dd9567f6715eb83ccb2 Mon Sep 17 00:00:00 2001
From: Xu Guangxin <guangxin.xu@intel.com>
Date: Thu, 7 May 2020 21:06:18 +0800
Subject: [PATCH 2/2] bufferpool: unblock acquire thread when we discard buffer

else the acquire thread will wait infinitely.
The deadlock showed in prevous unit test commit. This will fix it

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/470>
---
 gst/gstbufferpool.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gst/gstbufferpool.c b/gst/gstbufferpool.c
index 619860e63ff..8ae868cf2c7 100644
--- a/gst/gstbufferpool.c
+++ b/gst/gstbufferpool.c
@@ -1326,6 +1326,7 @@ not_writable:
 discard:
   {
     do_free_buffer (pool, buffer);
+    gst_poll_write_control (pool->priv->poll);
     return;
   }
 }
--
